tofu/geom/inputs/WEST_get_details_from_excel.py

```diff
@@ -0,0 +1,347 @@
+    },
+    'LDiv CasingCover': {
+        'name': 'CasingCoverLDivV0',
+        'class': 'PFC',
+        'save': True,
+    },
+    'Ldiv Casing': {
+        'name': 'CasingLDivV0',
+        'class': 'PFC',
+        'extent': _CASING_DTHETA,
+        'pos': _CASING_POS,
+        'save': True,
+    },
+    'Ldiv Casing PJ': {
+        'name': 'CasingPJLDivV0',
+        'class': 'PFC',
+        'extent': _PJ_DTHETA,
+        'pos': _PJ_POS,
+        'save': True,
+    },
+    'Ldiv PFU Plate': {
+        'name': 'CasingPFUPlateLDivV0',
+        'class': 'PFC',
+        'save': True,
+    },
+    # Upper divertor
+    'UDiv PFUs': {
+        'name': None,
+    },
+    'UDiv CasingCover': {
+        'name': 'CasingCoverUDivV0',
+        'class': 'PFC',
+        'save': True,
+    },
+    'Udiv Casing': {
+        'name': 'CasingUDivV0',
+        'class': 'PFC',
+        'extent': _CASING_DTHETA,
+        'pos': _CASING_POS,
+        'save': True,
+    },
+    'Udiv Casing PJ': {
+        'name': 'CasingPJUDivV0',
+        'class': 'PFC',
+        'extent': _PJ_DTHETA,
+        'pos': _PJ_POS,
+        'save': True,
+    },
+    'Udiv PFU Plate': {
+        'name': 'CasingPFUPlateUDivV0',
+        'class': 'PFC',
+        'save': True,
+    },
+}
+
+
+# #############################################################################
+# #############################################################################
+#           Define PEI
+# #############################################################################
+
+
+def get_PEI_HFS():
+
+    pts = np.array([
+        [-1681.880, 555.372, 784.274],
+        [-1670.349, 494.641, 778.897],
+        [-1661.932, 444.534, 774.972],
+        [-1652.797, 383.409, 770.712],
+        [-1642.850, 303.089, 766.074],
+        [-1635.270, 223.416, 762.539],
+        [-1630.704, 155.868, 760.410],
+        [-1627.334, 76.573, 758.838],
+        [-1626.250, 2.920, 758.333],
+        [-1627.273, -74.958, 758.810],
+        [-1630.443, -151.695, 760.288],
+        [-1635.199, -222.607, 762.506],
+        [-1641.223, -288.671, 765.315],
+        [-1649.738, -361.293, 769.285],
+        [-1662.010, -445.008, 775.008],
+        [-1670.139, -493.533, 778.799],
+        [-1681.880, -555.372, 784.274],
+    ])
+
+    poly = 1.e-3*np.array([np.hypot(pts[:, 0], pts[:, 2]), pts[:, 1]])
+    return poly
+
+
+def get_PEI_LFS(poly):
+
+    indsZ = np.argsort(poly[1, :])
+    poly = poly[:, indsZ]
+    poly_pplug = np.array([np.hypot(_PPLUG_PTS[:, 0], -_PPLUG_PTS[:, 2]),
+                           _PPLUG_PTS[:, 1]])
+    _, indspp = np.unique(poly_pplug[1, :], return_index=True)
+    poly_pplug = poly_pplug[:, indspp]
+    poly_pplug[0, :] = np.interp(poly_pplug[1, :], poly[1, :], poly[0, :])
+    ind_insert = np.searchsorted(poly[1, :], poly_pplug[1, :])
+
+    # TBF
+    poly_full = np.insert(poly, ind_insert, poly_pplug, axis=1)
+    pos_full_slim =  _PPLUG_POS[0] + 0.5*_DTHETA + None
+    extent_full = 
```

Comment:
This is the line that's making all jobs fail

Reply:
Yes, you're right, I realised too late there was something wrong.
I'm taking the opportunity to implement more details